client/src/components/ProtectedRoute.tsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { Route, RouteComponentProps, RouteProps, Redirect } from 'react-router-dom';
+import { useAuth } from '../context/useAuthContext';
+
+interface IProtectedRoute extends RouteProps {
```

Comment:
```suggestion
interface ProtectedRoute extends RouteProps {
```
It is recommended not to prefix interfaces with 'I' :)